aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/build/BuildTriggerStepRestartTest.java

```diff
@@ -60,18 +57,30 @@ public void evaluate() throws Throwable {
                           public void evaluate() throws Throwable {
                               story.j.waitUntilNoActivity();
                               FreeStyleProject p1 = story.j.jenkins.getItemByFullName("test1", FreeStyleProject.class);
-                              Run r = p1.getLastBuild();
+                              FreeStyleBuild r = p1.getLastBuild();
                               assertNotNull(r);
                               assertEquals(1, r.number);
                               assertEquals(Result.SUCCESS, r.getResult());
-                              assertEquals(0, story.j.jenkins.getQueue().getItems().length);
+                              assertFreeStyleProjectsInQueue(0);
                               WorkflowJob foo = story.j.jenkins.getItemByFullName("foo", WorkflowJob.class);
                               assertNotNull(foo);
-                              r = foo.getLastBuild();
-                              assertNotNull(r);
-                              story.j.assertBuildStatusSuccess(r);
+                              WorkflowRun r2 = foo.getLastBuild();
+                              assertNotNull(r2);
+                              story.j.assertBuildStatusSuccess(r2);
                           }
                       }
         );
     }
+
+    private void assertFreeStyleProjectsInQueue(int count) {
+        Queue.Item[] items = story.j.jenkins.getQueue().getItems();
```

Comment:
technically this **should** be asserting while holding the Queue lock... but in a test where decisions are not being made it is probably safe enough